### PR TITLE
Files for building conan package

### PR DESCRIPTION
**Pull request by Shaheen47**
Here is the new, clean pull request following pull request #69 .
This pull request addresses issue #68 .
The sources will be available to the importing project at `#include <cppitertools/*>` 

Please check the metadata in `conanfile.py` for correctness. 

For packaging and pushing the following steps are necessary:
1. Create an account following https://docs.conan.io/en/latest/uploading_packages/bintray/uploading_bintray.html . Assume your account name is **xyz**
2. package the package: `conan create . xyz/stable`
3. Follow the instructions for uploading: https://docs.conan.io/en/latest/uploading_packages/uploading_to_remotes.html

The package will be referred to as `cppitertools/2.0@xyz/stable`

Please refer to https://github.com/bigerl/testcppitertoolsconan for a example how to use the conan package.

### CODE REVIEW

File: conanfile.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+class Tentris_Parser(ConanFile):
+    name = "cppitertools"
+    version = "1.0"
+    author = "Ryan Haining"
+    description = "Implementation of python itertools and builtin iteration functions for C++17 "
+    topics = ("please add topics here")
+    settings = "build_type", "compiler", "os", "arch"
+    generators = "cmake", "cmake_find_package", "cmake_paths"
+
+    exports_sources = list()
```
Comment (ryanhaining) [inline conanfile.py]:
```
exports_sources = [f for f in os.listdir('.') if f.endswith('.hpp')]
```

the file name is already a string, list comprehension is shorter, `file` is a python builtin so you should not use it yourself.

File: conanfile.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+from conans import ConanFile, CMake
+
+import os
+
+
+class Tentris_Parser(ConanFile):
```
Comment (ryanhaining) [inline conanfile.py]: What is up with this class name?

File: conanfile.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    name = "cppitertools"
+    version = "1.0"
+    author = "Ryan Haining"
+    description = "Implementation of python itertools and builtin iteration functions for C++17 "
+    topics = ("please add topics here")
```
Comment (ryanhaining) [inline conanfile.py]: you need a trailing comma here to make this a tuple

File: conanfile.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    settings = "build_type", "compiler", "os", "arch"
+    generators = "cmake", "cmake_find_package", "cmake_paths"
+
+    exports_sources = list()
+    for file in os.listdir("."):
+        if file.endswith(".hpp"):
+            exports_sources.append(str(file))
+    print("found files: " + str(exports_sources))
```
Comment (ryanhaining) [inline conanfile.py]:
```
print('found files:', exports_sources)
```

File: conanfile.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    generators = "cmake", "cmake_find_package", "cmake_paths"
+    exports = "LICENSE.md"
+
+    exports_sources = list()
+    for file in os.listdir("."):
+        if file.endswith(".hpp"):
+            exports_sources.append(str(file))
+    print("found files: " + str(exports_sources))
+    exports_sources = tuple(exports_sources) + \
+                      ("internal/*", "CMakeLists.txt", "cmake/dummy-config.cmake.in")
```
Comment (bigerl) [inline conanfile.py]: Good idea. We could even put it all in one line making it a tuple directly:
```
exports_sources = tuple(f for f in os.listdir('.') if f.endswith('.hpp'))+ \
                      ("internal/*", "CMakeLists.txt", "cmake/dummy-config.cmake.in")
```

**Comment by ryanhaining** [comment]
I'll work on figuring out the rest

**Pull request merged**